src/test/kotlin/fr/social/gouv/agora/infrastructure/qag/repository/QagPopularRepositoryImplTest.kt

```diff
@@ -0,0 +1,127 @@
+package fr.social.gouv.agora.infrastructure.qag.repository
+
+import fr.social.gouv.agora.domain.QagInfo
+import fr.social.gouv.agora.domain.QagStatus
+import fr.social.gouv.agora.infrastructure.qag.dto.QagDTO
+import fr.social.gouv.agora.infrastructure.qag.repository.QagCacheRepository.CachePopularListResult
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.BDDMockito.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.util.*
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest
+internal class QagPopularRepositoryImplTest {
+
+    @Autowired
+    private lateinit var repository: QagPopularRepositoryImpl
+
+    @MockBean
+    private lateinit var databaseRepository: QagInfoDatabaseRepository
+
+    @MockBean
+    private lateinit var cacheRepository: QagCacheRepository
+
+    @MockBean
+    private lateinit var mapper: QagInfoMapper
+
+    private val qagDTO = QagDTO(
+        id = UUID.randomUUID(),
+        title = "title",
+        description = "description",
+        postDate = Date(42),
+        status = 256,
+        username = "username",
+        thematiqueId = UUID.randomUUID(),
+    )
+
+    private val qagInfo = QagInfo(
+        id = "id",
+        thematiqueId = "thematiqueId",
+        title = "title",
+        description = "description",
+        date = Date(14),
+        status = QagStatus.MODERATED,
+        username = "username",
+    )
+
+    @Test
+    fun `getQagPopularList - when cache is not initialized - should call getQagPopularListFromDatabase and insert result (case result = emptylist) to cache`() {
```

Comment:
Euh c'était pas tout à fait ce que je voulais dire, je voyais un changement du genre:
`getQagPopularList - when CacheNotInitialized & database return emptyList - should call getQagPopularListFromDatabase and insert result to cache`

Reply:
done